app/controllers/user_api_key_clients_controller.rb

```diff
@@ -2,26 +2,36 @@
 class UserApiKeyClientsController < ApplicationController
   layout "no_ember"
 
-  requires_login
-  skip_before_action :check_xhr, :preload_json
+  skip_before_action :check_xhr, :preload_json, :verify_authenticity_token
```

Comment:
Is it possible at all to keep this limited to authenticated requests? I get why we may wish to open this up to anons, the client row may need to be created by an unauthenticated third party. I'm a bit concerned though that this opens up a potential attack vector, a carefully crafted request could create dummy clients. It's mitigated by scopes to some extent...

I guess a malicious actor could also create a user and exploit this that way as well. But at least there would be a bit more tracing.

Reply:
I did a fair bit of thinking on this this week, and came up with the following: https://github.com/angusmcleod/discourse/commit/a87176a39b3547162fc012d7aa4c5b056a4815f0

Adds:
- a relatively strict rate-limit for the "create" action.
- Only permits creation, not updates. This ensures that the public key and auth_redirect associated with a client always remains the same. If you want to change these for a "registered" client you have to create a new one with a different id.
- A cleanup job to handle "stale" clients (where all the client's keys are older than x days).
- Make the `allow_user_api_key_client_scopes` site setting hidden, to reduce the chance of an admin adding overly broad scopes to this allow-list. Moreover this functionality, i.e. 1:many will likely always be managed programmatically anyway.

Does that ally your concerns here?

Reply:
Yes, thank you! That does indeed address my concerns.